app/src/main/kotlin/at/bitfire/davdroid/webdav/DavDocumentsProvider.kt

```diff
@@ -127,6 +128,44 @@ class DavDocumentsProvider(
         fun notifyMountsChanged(context: Context) {
             context.contentResolver.notifyChange(buildRootsUri(context.getString(R.string.webdav_authority)), null)
         }
+
+        /**
+         * Contains a map that matches the column names from the documents provider and Room.
+         */
+        private val columnsMap = mapOf(
+            Document.COLUMN_DOCUMENT_ID to "id",
+            Document.COLUMN_DISPLAY_NAME to "displayName",
+            Document.COLUMN_MIME_TYPE to "mimeType",
+            Document.COLUMN_SIZE to "size",
+            Document.COLUMN_LAST_MODIFIED to "lastModified",
+        )
+
+        @VisibleForTesting
+        fun processOrderByQuery(orderBy: String): String? {
```

Comment:
What the method does is good, but I'd really prefer a positive way of constructing dangerous SQL clauses, that is not to rewrite an existing String so that it fits, but to create it from scratch. Like:

1. Start with empty result.
1. Parse incoming orderBy.
1. For each parsed token: if we have a mapping, append the mapping (which is defined by us) to result. If we don't have mapping, log a warning and ignore.
1. At the end, join the result and return it.

We can then be sure that there's nothing we have overlooked when processing the string.